### ISSUE TITLE
Tx assign-consensus-key is not possible with ledger

### ISSUE THREAD

**Issue by marcotradenet** [cosmos/gaia#2469]
## Summary of Bug

Trying to assign a new node key for a consumer chain using the Nano Ledger I get a Panic

## Version

v9.0.3

## Steps to Reproduce

1. Created a node for the consumer chain `Neutron` 
2. Get pub_key3. 
   ```bash
   neutrond tendermint show-validator
   # {"@type":"/cosmos.crypto.ed25519.PubKey","key":"dz1HakYtnfeP4gelfAVPtHw6M19VdsQrvnk8r9f8EmQ="}
   ```
3. Try assign the pub_key
   ```bash
   NEUTRON_KEY='{"@type":"/cosmos.crypto.ed25519.PubKey","key":"dz1HakYtnfeP4gelfAVPtHw6M19VdsQrvnk8r9f8EmQ="}'
   WALLET="cosmos_operator_wallet"
   CHAINID="cosmoshub-4"
   
   gaiad \
     tx provider assign-consensus-key neutron-1 \
     $NEUTRON_KEY \
     --keyring-backend=file \
     --sign-mode=amino-json \
     --from $WALLET \
     --ledger \
     --fees 500uatom \
     --chain-id $CHAINID -y
   ```
4. Obtain the error
```
panic: unable to resolve type URL /cosmos.crypto.ed25519.PubKey

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/codec.(*ProtoCodec).MustMarshalJSON(...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/codec/proto_codec.go:147
github.com/cosmos/interchain-security/x/ccv/provider/types.MsgAssignConsumerKey.GetSignBytes({{0x7ffeefbff8d3, 0x9}, {0xc0001f0b00, 0x34}, 0xc001209b00})
	github.com/cosmos/interchain-security@v1.1.0/x/ccv/provider/types/msg.go:62 +0xd8
github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx.StdSignBytes({0x7ffeefbff9e5, 0xb}, 0x1b32, 0x482, 0x0, {{0xc0001ae648?, 0x1f?, 0xc0012a60d8?}, 0x10005379c?}, {0xc0001e65f0, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/x/auth/legacy/legacytx/stdsign.go:61 +0x284
github.com/cosmos/cosmos-sdk/x/auth/tx.signModeLegacyAminoJSONHandler.GetSignBytes({}, 0x53310?, {{0x7ffeefbff9e5?, 0xc0012a61e0?}, 0x10000df05?, 0xc0012a61a0?}, {0x10221df98?, 0xc000124e80?})
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/x/auth/tx/legacy_amino_json.go:49 +0x1a7
github.com/cosmos/cosmos-sdk/x/auth/signing.SignModeHandlerMap.GetSignBytes({0x1, {0xc00111f668, 0x2, 0x2}, 0xc001180540}, 0x2231450?, {{0x7ffeefbff9e5, 0xb}, 0x1b32, 0x482}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/x/auth/signing/handler_map.go:59 +0xa2
github.com/cosmos/cosmos-sdk/client/tx.Sign({{0x102244250, 0xc000124cc0}, {0x10223c780, 0xc00110c9c0}, {0x102233318, 0x1031008d8}, 0x1b32, 0x482, 0x30d40, 0x0, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/client/tx/tx.go:388 +0x389
github.com/cosmos/cosmos-sdk/client/tx.BroadcastTx({{0xc0011e01b0, 0x14, 0x18}, {0x1022487a8, 0xc00124ec90}, {0x7ffeefbff9e5, 0xb}, {0x102236c50, 0xc00114d9e0}, {0x102242c28, ...}, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/client/tx/tx.go:131 +0x68d
github.com/cosmos/cosmos-sdk/client/tx.GenerateOrBroadcastTxWithFactory({{0xc0011e01b0, 0x14, 0x18}, {0x1022487a8, 0xc00124ec90}, {0x7ffeefbff9e5, 0xb}, {0x102236c50, 0xc00114d9e0}, {0x102242c28, ...}, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/client/tx/tx.go:50 +0x11f
github.com/cosmos/interchain-security/x/ccv/provider/client/cli.NewAssignConsumerKeyCmd.func1(0xc001254600?, {0xc000e98540, 0x2, 0xe?})
	github.com/cosmos/interchain-security@v1.1.0/x/ccv/provider/client/cli/tx.go:60 +0x465
github.com/spf13/cobra.(*Command).execute(0xc001254600, {0xc000e98460, 0xe, 0xe})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000f1f200)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.6.1/command.go:961
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x0?, {0xc0005a78c0, 0x12})
	github.com/cosmos/cosmos-sdk@v0.45.15-ics/server/cmd/execute.go:36 +0x1eb
main.main()
	github.com/cosmos/gaia/v9/cmd/gaiad/main.go:16 +0x2c
```
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

**Comment by mkaczanowski** [comment]
I can confirm, I see the same issue. Can this be fixed before neutron genesis?

**Comment by mkaczanowski** [comment]
I managed to bypass the issue. Pasting for people who are in rush due to neutron launch. (this is for github.com/cosmos/interchain-security repo):

```
diff --git a/x/ccv/provider/types/msg.go b/x/ccv/provider/types/msg.go
index 55c0bd40..a4605c03 100644
--- a/x/ccv/provider/types/msg.go
+++ b/x/ccv/provider/types/msg.go
@@ -59,7 +59,7 @@ func (msg MsgAssignConsumerKey) GetSigners() []sdk.AccAddress {
 
 // GetSignBytes returns the message bytes to sign over.
 func (msg MsgAssignConsumerKey) GetSignBytes() []byte {
-       bz := ModuleCdc.MustMarshalJSON(&msg)
+       bz := AminoCdc.MustMarshalJSON(&msg)
        return sdk.MustSortJSON(bz)
 }
 

```

UPDATE: It will let you sign on the ledger, but once TX is broadcasted the node will fail to verify signature. So the actual fix needs to be implemented and rolled out. Alternative is to derive a sub account with authz and sign from there

**Comment by MattKetmo** [comment]
Got the issue too. This is now fixed with version `9.1.0`
